### ISSUE TITLE
[Feature]: green-cost metrics with the Ecologits observabilty plugin for litellm proxy

### ISSUE THREAD

**Issue by Jesshuan** [BerriAI/litellm#31226]
### Check for existing issues

- [x] I have searched the existing issues and checked that my issue is not a duplicate.

### The Feature

We propose to enrich litellm's existing llm call metrics via an optionnal Ecologits plugin that will provide green-cost metrics, based on the [Ecologits API](https://api.ecologits.ai/docs) calculator, and for visibility through all the downstream observability callback plugin: Datadog, Langfuse, etc.

For this first raw 'proxy mode' functional version, the principle is simple:

- when a new model is registered, the litellm admins can add a model info argument untitled "ecologits_electricity_mix_zone" (e.g. 'FRA', 'DEU') for this model. If they do not, the default electrify mix zone call back simply to the default zone "WOR" (for "world").

- for each successfull call from litellm to an llm provider, the green impact is evaluated with an API call to Ecologits API (or the internal Ecologits API self-hosted by the enterprise) and these new green metric enrich the initial payload for all the litellm supported callback integrations plugin, choosen by the users: langfuse, datadog, prometheus metrics ... For this latest, some green-cost metrics counter are defined and exposed, side-to-side to the payload enrichement.

ecologits_electricity_mix_zone

### Motivation, pitch

For the litellm users, it is not enough to make llm deployments decisions with quota per teams for billing, without take into account the ecological cost of their decisions. Moreover, for many organization, it is not a choice, it's a new reglementation compliance. Many reglementations will arrive to ask for the Data and IA department to evaluate their green cost impact for their llm calls.

**Ecologits**(https://ecologits.ai/latest/) is a non-profit project, supported by different institutional and non-institutional partners and hosted by [CodeCarbon](https://codecarbon.io/) a famous non-profit organization that provide tools for green-cost evaluation for IT.
You can see the methodology applied by the Ecologits API for llm inference [here](https://ecologits.ai/latest/methodology/llm_inference/ ). Evaluating the green-cost of these llm calls is not an eass task, in a world where the main providers of the main big models do not reaveal their internal llm architecture... But, for my point of view, this methodology is at a crossing between reasearch paper estimation (at the state of the art), real data-center experiments, and ML Engineering and Ops common usages, that we could suppose to be real, although some limitations remain.

With a new callback integration with Ecologits, the green-cost calculation for llm inference is decoupled for litellm itself. This allows to the litellm developper to not maintain an efficient calculator at the state of the art, and to benefit from the future improvement of the Ecologit API.
Of course, this integration could be coexists with any future native Litellm green-cost evaluation.

### What part of LiteLLM is this about?

Proxy

### LiteLLM is hiring a founding backend engineer, are you interested in joining us and shipping to all our users?

No

### Twitter / LinkedIn details

https://www.linkedin.com/in/jesshuan-din%C3%A9-a1092556
